src/main/java/picard/illumina/CheckIlluminaDirectory.java

```diff
@@ -309,6 +314,9 @@ private static int verifyLane(final IlluminaFileUtil fileUtil, final List<Intege
 
         for (final IlluminaFileUtil.SupportedIlluminaFormat format : formatToDataTypes.keySet()) {
             final ParameterizedFileUtil util = fileUtil.getUtil(format);
+            if(util instanceof PerTileOrPerRunFileUtil){
```

Comment:
Is there any way to avoid `instanceof` here by using an overridden method? E.g. make this method call `setTilesForXXXOnly()` (or whatever name makes sense) and only implement it in the subclass.